salt/states/chocolatey.py

```diff
@@ -0,0 +1,162 @@
+
+    force_x86
+      Force x86 (32bit) installation on 64 bit systems. Defaults to false.
+
+
+    .. code-block:: yaml
+
+        Installsomepackage:
+          chocolatey.install:
+            - name: packagename
+            - version: '12.04'
+            - source: 'mychocolatey/source'
+            - force: True
+
+
+
+    '''
+
+    ret = {'name': name,
+           'result': True,
+           'changes': {},
+           'comment': ''}
+
+    # Determine if the package is installed
+    if name not in __salt__['cmd.run']('choco list --local-only'):
+        ret['changes'] = {'name': '{0} will be installed'.format(name)}
+    elif force:
+        ret['changes'] = {'name': '{0} is already installed but will reinstall'
+            .format(name)}
+    else:
+        ret['comment'] = 'The Package {0} is already installed'.format(name)
+        return ret
+
+    if __opts__['test']:
+        ret['result'] = None
+        return ret
+
+    # Install the package
+    ret['changes'] = {name: __salt__['chocolatey.install'](name, version,
+                                                             source,
+                                                             force,
+                                                             install_args,
+                                                             override_args,
+                                                             force_x86)}
+
+    if 'Running chocolatey failed' not in ret['changes']:
+        ret['result'] = True
+    else:
+        ret['result'] = False
+
+    if not ret['result']:
+        ret['comment'] = 'Failed to install the package {0}'.format(name)
+
+    return ret
+
+
+def uninstall(name, version=None, uninstall_args=None, override_args=False):
+    '''
+    Uninstalls a package
+
+    name
+      The name of the package to be uninstalled
+
+    version
+      Uninstalls a specific version of the package. Defaults to latest
+      version installed.
+
+    uninstall_args
+      A list of uninstall arguments you want to pass to the uninstallation
+      process i.e product key or feature list
+
+    override_args
+      Set to true if you want to override the original uninstall arguments (
+      for the native uninstaller)in the package and use your own.
+      When this is set to False uninstall_args will be appended to the end of
+      the default arguments
+
+
+
+
+    .. code-block: yaml
+
+      Removemypackage:
+        chocolatey.uninstall:
+          - name: mypackage
+          - version: '21.5'
+
+    '''
+
+    ret = {'name': name,
+           'result': True,
+           'changes': {},
+           'comment': ''}
+
+    # Determine if package is installed
+    if name in __salt__['cmd.run']('choco list --local-only'):
+        ret['changes'] = {'name': '{0} will be removed'.format(name)}
+    else:
+        ret['comment'] = 'The package {0} is not installed'.format(name)
+        return ret
+
+    if __opts__['test']:
+        ret['result'] = None
```

Comment:
Same thing here. We need to inform the user what *would* have happened instead of just doing a no-op.